packages/core/src/components/tabs2/tabs.tsx

```diff
@@ -0,0 +1,285 @@
+
+    /**
+     * Whether to show tabs stacked vertically on the left side.
+     * @default false
+     */
+    vertical?: boolean;
+
+    /**
+     * A callback function that is invoked when tabs in the tab list are clicked.
+     */
+    onChange?(newTabId: TabId, prevTabId: TabId): void;
+}
+
+export interface ITabs2State {
+    indicatorWrapperStyle?: React.CSSProperties;
+    selectedTabId?: TabId;
+}
+
+@PureRender
+export class Tabs2 extends AbstractComponent<ITabs2Props, ITabs2State> {
+    public static Tab = Tab2;
+
+    public static defaultProps: ITabs2Props = {
+        animate: true,
+        id: "",
+        renderActiveTabPanelOnly: false,
+        vertical: false,
+    };
+
+    public displayName = "Blueprint.Tabs2";
+
+    private tabElement: HTMLDivElement;
+
+    constructor(props?: ITabs2Props, context?: any) {
+        super(props, context);
+        let selectedTabId = props.selectedTabId;
+        if (selectedTabId === undefined) {
+            // select first tab in absence of user input
+            selectedTabId = props.defaultSelectedTabId === undefined
+                ? this.getTabChildren()[0].props.id
+                : props.defaultSelectedTabId;
+            // NOTE: providing an unknown ID will hide the selection
+        }
+        this.state = { selectedTabId };
+    }
+
+    public render() {
+        const { indicatorWrapperStyle, selectedTabId } = this.state;
+
+        const tabTitles = React.Children.map(this.props.children, (child) => {
+            if (isTab(child)) {
+                const { id } = child.props;
+                return (
+                    <TabTitle
+                        {...child.props}
+                        parentId={this.props.id}
+                        onClick={this.getTabClickHandler(id)}
+                        selected={id === selectedTabId}
+                    />
+                );
+            } else {
+                // TabTitle renders an <li> so let's do the same here
+                return <li>{child}</li>;
+            }
+        });
+
+        const tabPanels = this.getTabChildren()
+            .filter(this.props.renderActiveTabPanelOnly ? (tab) => tab.props.id === selectedTabId : () => true)
+            .map(this.renderTabPanel);
+
+        const tabIndicator = (
+            <div className="pt-tab-indicator-wrapper" style={indicatorWrapperStyle}>
+                <div className="pt-tab-indicator" />
+            </div>
+        );
+
+        const classes = classNames(Classes.TABS, { "pt-vertical": this.props.vertical }, this.props.className);
+        return (
+            <div className={classes}>
+                <div
+                    className={Classes.TAB_LIST}
+                    onKeyDown={this.handleKeyDown}
+                    onKeyPress={this.handleKeyPress}
+                    ref={this.handleTabRef}
+                    role="tablist"
+                >
+                    {this.props.animate ? tabIndicator : undefined}
```

Comment:
If `animate={false}`, do we still show a blue underline beneath the selected tab?